src/components/dogDetail.jsx

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+
+class DogDetail extends Component {
+  state = {
+    dog: null
+  }
+
+  componentDidMount() {
+    console.log(this.props.location.state);
+    this.setState({ dog: this.props.location.state.dog });
+  }
+
+  render() {
+    const { dog } = this.props.location.state
```

Comment:
Toto by sme si na najblizsom calle mohli prebrat - ako pristupit k datam pre jeden objekt - psa. Ako na to pristupit cez `Route`.